airsonic-main/src/main/java/org/airsonic/player/service/MediaScannerService.java

```diff
@@ -279,20 +311,43 @@ else if (file.isAudio()) {
         }
     }
 
-    private void updateAlbum(MediaFile file, MusicFolder musicFolder, Instant lastScanned, Map<String, Integer> albumCount) {
+    private void updateAlbum(MediaFile file, MusicFolder musicFolder, Instant lastScanned, Map<String, AtomicInteger> albumCount, Map<MediaFile, Album> albums) {
         String artist = file.getAlbumArtist() != null ? file.getAlbumArtist() : file.getArtist();
         if (file.getAlbumName() == null || artist == null || file.getParentPath() == null || !file.isAudio()) {
             return;
         }
+        
+        final AtomicBoolean firstEncounter = new AtomicBoolean(false);
+        Album album = albums.compute(file, (k,v) -> {
+            Album a = v;
+            
+            if (a == null) {
+                a = albumDao.getAlbumForFile(k);
+            }
+            
+            if (a == null) {
+                a = new Album();
+                a.setPath(k.getParentPath());
+                a.setName(k.getAlbumName());
+                a.setArtist(artist);
+                a.setCreated(k.getChanged());
+            }
+            
+            firstEncounter.set(!lastScanned.equals(a.getLastScanned()));
+            
+            if (file.getDurationSeconds() != null) {
+                a.incrementDurationSeconds(file.getDurationSeconds());
+            }
+            if (file.isAudio()) {
+                a.incrementSongCount();
+            }
+            
+            a.setLastScanned(lastScanned);
+            a.setPresent(true);
+            
+            return a;
+        });
```

Comment:
This is guaranteed to be synchronized for the same object on the ConcurrentHashMap